spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/support/JdbcRepositoryFactoryBean.java

```diff
@@ -141,17 +166,70 @@ public void afterPropertiesSet() {
 		Assert.state(this.mappingContext != null, "MappingContext is required and must not be null!");
 		Assert.state(this.converter != null, "RelationalConverter is required and must not be null!");
 
-		if (dataAccessStrategy == null) {
-
-			SqlGeneratorSource sqlGeneratorSource = new SqlGeneratorSource(mappingContext);
-			this.dataAccessStrategy = new DefaultDataAccessStrategy(sqlGeneratorSource, mappingContext, converter,
-					operations);
-		}
+		ensureJdbcOperationsIsInitialized();
+		ensureDataAccessStrategyIsInitialized();
 
 		if (queryMappingConfiguration == null) {
 			this.queryMappingConfiguration = QueryMappingConfiguration.EMPTY;
 		}
 
 		super.afterPropertiesSet();
 	}
+
+	private void ensureJdbcOperationsIsInitialized() {
+
+		if (operations != null) {
+			return;
+		}
+
+		operations = findBeanByNameOrType(NamedParameterJdbcOperations.class, jdbcOperationsRef);
+
+	}
+
+	private void ensureDataAccessStrategyIsInitialized() {
+
+		if (dataAccessStrategy != null) {
+			return;
+		}
+
+		dataAccessStrategy = findBeanByNameOrType(DataAccessStrategy.class, dataAccessStrategyRef);
```

Comment:
This method is a good candidate for `BeanFactory.getBeanProvider(…).getIfAvailable(Supplier)`